VolumeLogic/Application.py

```diff
@@ -20,11 +21,12 @@ def name(self, name):
 
     @property
     def volume(self):
-        return self.volume
+        return self.os_connection.getProcessVolume(self.name)
 
     @volume.setter
     def volume(self, value):
-        self.volume = VolumeCalculation.convert_to_volume(self.volume, value)
+        volume = VolumeCalculation.convert_to_volume(self.volume, value)
```

Comment:
Is `self.volume` going to exist here? You removed it a few lines up.

Reply:
That is actually a call to the getter. The @property tags try and make it so that python still looks like python (you don't need to put () around the function call) when you get the value. I will add _ to the other variables though so that it is more clear what is happening (_ in front of a variable means private). https://www.geeksforgeeks.org/getter-and-setter-in-python/

```
# Python program showing a 
# use of property() function 

class Geeks: 
	def __init__(self): 
		self._age = 0
	
	# function to get value of _age 
	def get_age(self): 
		print("getter method called") 
		return self._age 
	
	# function to set value of _age 
	def set_age(self, a): 
		print("setter method called") 
		self._age = a 

	# function to delete _age attribute 
	def del_age(self): 
		del self._age 
	
	age = property(get_age, set_age, del_age) 

mark = Geeks() 

mark.age = 10

print(mark.age) 
```